src/code/z_player_lib.c

```diff
@@ -2302,56 +3398,424 @@ Vec3f D_801C0EA0 = { 1300.0f, -400.0f, 0.0f };
-        Matrix_MultVec3f(iter, &sp84);
-        Matrix_MultVec3f(iter2, &sp78);
+                CLOSE_DISPS(play->state.gfxCtx);
+            } else if (&gPlayerAnim_pg_punchB == player->skelAnime.animation) {
+                func_80127488(play, player, D_801C07AC[(s32)player->skelAnime.curFrame]);
+            } else {
+                func_801271B0(play, player, 1);
+            }
+        }
 
-        func_80127DA4(play, D_801F58B0[i], D_801C0C54, 3, &sp84, &sp78, &sp6C);
-        sp6C += 11;
+        if (player->actor.scale.y >= 0.0f) {
+            if (player->rightHandType == 0xFF) {
+                Matrix_Get(&player->shieldMf);
+            } else if (player->rightHandType == PLAYER_MODELTYPE_RH_SHIELD) {
+                Matrix_Get(&player->shieldMf);
+                func_801265C8(play, player, &player->shieldQuad, D_801C0DA8);
+            } else if (player->rightHandType == PLAYER_MODELTYPE_RH_HOOKSHOT) {
+                Matrix_MultVec3f(&D_801C0DD8, &player->rightHandWorld.pos);
+
+                if (sp224 != NULL) {
+                    MtxF sp1BC;
+
+                    Matrix_MultVec3f(&D_801C0DE4, &sp224->world.pos);
+                    Matrix_RotateZYX(0, -0x4000, -0x4000, MTXMODE_APPLY);
+                    Matrix_Get(&sp1BC);
+                    Matrix_MtxFToYXZRot(&sp1BC, &sp224->world.rot, 0);
+                    sp224->shape.rot = sp224->world.rot;
+                    if (func_800B7128(player) != 0) {
+                        Matrix_Translate(500.0f, 300.0f, 0.0f, MTXMODE_APPLY);
+                        func_80124CC4(play, player, 77600.0f);
+                    }
+                }
+            } else if ((player->meleeWeaponState != 0) &&
+                       (player->meleeWeaponAnimation == PLAYER_MWA_GORON_PUNCH_RIGHT)) {
+                func_80126B8C(play, player);
+            }
+            if ((player->getItemDrawId != 0) || ((func_800B7118(player) == 0) && (sp224 != NULL))) {
+                if (!(player->stateFlags1 & PLAYER_STATE1_400) && (player->getItemDrawId != 0) &&
+                    (player->exchangeItemId != PLAYER_IA_NONE)) {
+                    Math_Vec3f_Copy(&D_801F59E8, &player->leftHandWorld.pos);
+                } else {
+                    D_801F59E8.x =
+                        (player->bodyPartsPos[PLAYER_BODYPART_RIGHT_HAND].x + player->leftHandWorld.pos.x) * 0.5f;
+                    D_801F59E8.y =
+                        (player->bodyPartsPos[PLAYER_BODYPART_RIGHT_HAND].y + player->leftHandWorld.pos.y) * 0.5f;
+                    D_801F59E8.z =
+                        (player->bodyPartsPos[PLAYER_BODYPART_RIGHT_HAND].z + player->leftHandWorld.pos.z) * 0.5f;
+                }
 
-        Matrix_Push();
-        Matrix_Translate(iter->x, iter->y, iter->z, MTXMODE_APPLY);
-        func_80128388(D_801F58B0[i], D_801C0C54, 3, &mtx);
-        Matrix_Pop();
-        iter++;
-        iter2++;
-    }
+                if (player->getItemDrawId == 0) {
```

Comment:
```suggestion
                if (player->getItemDrawId == (GID_NONE + 1)) {
```
Pretty sure this is actually `getItemDrawIdPlusOne`, not sure if that's commented on or if the name should change

Reply:
Whooooops... That explains all the nonsense checks in `z_player.c` :sweat_smile: 
I'll rename it